src/dirwatcher.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const fsPromises = require('fs/promises');
+const { EventEmitter } = require('events');
+
+class DirWatcher extends EventEmitter {
+  constructor() {
+    super();
+    this.directoryContents = {};
+    this.newDirectoryContents = {};
+    this.changedFiles = [];
+  }
+
+  async readDir(path) {
+    return fsPromises.readdir(path);
+  }
+
+  async readStats(path, fileNames) {
+    return fileNames.map(fileName => fsPromises.stat(`${path}/${fileName}`));
+  }
+
+  async saveState(path) {
+    // read dir file names
+    const fileNames = await this.readDir(path);
+    this.changedFiles = fileNames;
+    this.directoryContents > 0
+      ? this.emit('data change')
+      : this.emit('no data');
+    // read stats
+    const stats = await this.readStats(path, fileNames);
+    const statsResolved = await Promise.all(stats);
+
+    // combine stats
+    const statsArray = fileNames.map((el, index) => ({
+      [el]: statsResolved[index].mtime.toString().slice(0, 24),
```

Comment:
If you will use `Map` instead of plain object, it will be better to use `number` instead of `string`, so you can use timestamp as a key for `Map`